Maps trailloading

mikkokotola opened the pull request:
Trail data import from xml-files and trail data -related classes. All new classes are tested through the importer class tests.

laitalaj commented:
Tests were successful in staging: https://developer.cloud.unity3d.com/build/orgs/jiilaitala/projects/3dmaps/buildtargets/staging-android/builds/14/log/

laitalaj commented:
A few changes that you guys could do before we merge this:

* Getter and setter naming in TrailNode: we've been doing UpperCamelCase
* TrailNodeIdCorrect-test: check that the trail contains all the required nodes (just checking one node doesn't make sure that every node is included)
* "magic number" string literals in node type switch and child node naming: instead of "magic number" string literals in code, do for example `private const string wayType = "way", nodeType = "node", childNodeType = "nd";` for easier maintenance and better reusability

@mikkokotola @Namirual

mikkokotola commented:
Good points! The tests already checked both the first trail's first node and (in the latlon-test) the last trail's last node, so I guess would have already been quite OK, but I added the requested checking of all node id's for the first trail. Fixed the other parts where Julius identified problems. Should be OK now!
